src/test/scala/reporter/RemoraDatadogReporterSpec.scala

```diff
@@ -0,0 +1,39 @@
+package reporter
+
+
+import com.codahale.metrics.{Metric, MetricFilter, MetricRegistry}
+import config.DataDog
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers, PrivateMethodTester}
+
+class RemoraDatadogReporterSpec extends FlatSpec with Matchers with PrivateMethodTester with MockFactory {
+
+  private val metricRegistry: MetricRegistry = new MetricRegistry
+  private val metric: Metric = mock[Metric]
+
+  "Metrics filter" should "match any metric when no filter is given" in {
+    val filter = buildMetricFilter(List.empty)
+
+    filter.matches("any_metrics_name", metric) should be(true)
+  }
+
+  it should "match metric containing consumer group name" in {
+    val kafkaConsumerGroupName = "test-consumer1"
+    val filter = buildMetricFilter(List(kafkaConsumerGroupName))
+
+    filter.matches(s"metric-name-$kafkaConsumerGroupName", metric) should be(true)
+  }
+
+  it should "not match metric containing consumer group name" in {
+    val filter = buildMetricFilter(List("test-consumer1"))
+
+    filter.matches("some-metrics", metric) should be(false)
+  }
+
+
+  private def buildMetricFilter(kafkaConsumerList: List[String]): MetricFilter = {
+    val config = DataDog(enabled = true, "test", 1, "localhost", 8125, kafkaConsumerList)
+    val reporter = new RemoraDatadogReporter(metricRegistry, config)
+    reporter invokePrivate PrivateMethod[MetricFilter]('kafkaConsumerGroupFilter)()
```

Comment:
you sneaky dog you :)

Reply:
Ahah, totally agree about this :) 
This filter is basically the only logic/source of errors that I added, so I figured I could test it.
I could also run some kind of integration tests, but I am not sure about that.
What do you think?

Reply:
Integration tests would be cool but I think this is for a larger conversation. There is no integration/acceptance tests framework....i'll have to add it in later.

For now, i think the logic is tested and once you gave a manual test to make sure it's working as expected then lets get it in.

Reply:
Sure, this code is running on our setup since some weeks already.